libats-http/src/main/scala/com/advancedtelematic/libats/http/ErrorHandler.scala

```diff
@@ -16,46 +16,60 @@ import com.advancedtelematic.libats.data.{ErrorCode, ErrorCodes, ErrorRepresenta
 import de.heikoseeberger.akkahttpcirce.FailFastCirceSupport._
 import io.circe.syntax._
 import com.advancedtelematic.libats.codecs.CirceUuid._
-import io.circe.Json
+import io.circe.{Encoder, Json}
+import cats.syntax.option._
 
 import scala.reflect.ClassTag
 import scala.util.control.NoStackTrace
+import scala.language.existentials
 
 object Errors {
   import Directives._
   import ErrorRepresentation._
 
-  import scala.language.existentials
+  abstract class Error[T](val code: ErrorCode,
+                          val responseCode: StatusCode,
+                          val msg: String,
+                          val cause: Throwable = null,
+                          val errorId: UUID = UUID.randomUUID()) extends Throwable(msg, cause) with NoStackTrace
```

Comment:
This `errorId` and `cause` seems to not be used in the handler below?

Reply:
👍